### DISCUSSION TITLE
Issues Details Page: Straightening Out the Kinks - Tell Us What You Think (+More!)

### DISCUSSION THREAD

**Discussion by therealarkin** [getsentry/sentry#43531] [Problem Understanding]
We want to extend a big thank you to everyone who participated in our recent Developer satisfaction survey for the Sentry Issues details page. Your feedback is incredibly valuable to us as we work to improve the Sentry experience for all of our Developers. We wanted to share some of the positive and negative feedback we received from the survey and what we are doing about it. Here are a few examples (verbatim):

**Positive** (this helps keep the team motivated!)
- "You guys are great. thank you"
- "Lots of useful information very well organized"
- "It's familiar and I've learned where everything is over the years"
- "It does most of what I want"
- "I would like to be able to see a little more code above and below"

**Negative** (this helps keep the team employed, lol):
- "Too much information for 1 page. The breadcrumbs table is way to small, it is hard to check it"
- “Very good in general, except the breadcrumbs panel that is really really too short”
- "Issues are hard to read, the raw stacktrace is usually more useful. Suggested assignees aren't even team members in Sentry anymore"

We're happy to share that we quickly shipped an improvement to breadcrumb visibility [check out this [video](https://drive.google.com/file/d/1-9MN1ibtboR4cajfoPzplS9OQWpW0sYV/view)] This addresses one of the most common feedback.  
Additionally, we're also looking into cases where data was missing when source mapping was enabled and, in general, making source mapping & suggested assignees much better.

New-to-Sentry Developers found it more challenging to use the issue details page. To help us continue to improve the user experience for everyone and, in particular New-to-Sentry Developers, we would appreciate you providing specific examples of things you found difficult to do when using Sentry for the first few times and how you overcame them. To get the conversation going, please help us with the following:

1. What was the most challenging thing you encountered when using the Issue details page for the first few times?
2. What helped you overcome those challenges?
3. What suggestions do you have for making the Issue details page more intuitive for new-to-Sentry Developers?
4. How can the issue details page experience be more actionable? What should be added? What should be removed? And why?

**Comment by mwarkentin** [comment]
A couple of random thoughts.

### Better temporal control

I'd love to be able to have more flexible views of time for the issue I'm investigating.

An example:

![image](https://user-images.githubusercontent.com/67560/213841285-682fb335-b4ac-4490-8225-9d0948430137.png)

It would be great to be able to click on that 1/24 slice in the graph and zoom into that with a new visualization with per minute or per-5 minute rates, update the tag breakdowns so I can see if it was a specific browser or release or whatever that triggered a spike.

### Friendly JSON view

That big JSON link at the top of the issue page just dumps me into a massive unreadable text file:

![image](https://user-images.githubusercontent.com/67560/213841437-780c34d9-7f34-40fe-8e81-94692c95c729.png)

![image](https://user-images.githubusercontent.com/67560/213841412-93da7d96-47b9-4eac-81d7-bccb3ebe117c.png)

It would be great if that dropped me into something more user-friendly for exploring the JSON (think a hosted JSON UI like [JSONHero](https://jsonhero.io). Of course there should still be an easy way to get to the raw JSON for use in other tools.

On that note the even bigger link with the event ID doesn't seem to actually do anything? Seems like it should at least give me a quick "copy permalink to event" or something.

### Event Grouping UI Issues

Large numbers of event grouping information seem to overflow the page which doesn't look great:

![image](https://user-images.githubusercontent.com/67560/213841573-b3082e0c-9ca6-4538-af3a-fee32ec8145b.png)

And even worse when I scroll over to view it somehow it messes with my browser and I can't scroll back normally (stuck here):

![image](https://user-images.githubusercontent.com/67560/213841617-9a9a2938-29f1-4b26-a886-c65a13c9f963.png)

I was able to scroll back by selecting text and "dragging" the page (using Safari 16.2):

![event-grouping-jank](https://user-images.githubusercontent.com/67560/213841666-f522b35c-0054-49ed-986b-f3b2413eaab7.gif)

### Tags / Context overlap

We seem to have some (but not all) duplicate information in both tags and the "context" information lower in the page. Example: `organization` and `organization.slug` in tags and organization id and slug in context:

![image](https://user-images.githubusercontent.com/67560/213841784-8bb3398c-7607-489b-bc4b-dc18a7615ec5.png)

![image](https://user-images.githubusercontent.com/67560/213841803-4fc24202-d8d0-4312-bb32-6c1b5f16fa65.png)

In addition the same information is clickable as a tag which takes me to a custom search, but in the context area it's just plain text - feels inconsistent.

![image](https://user-images.githubusercontent.com/67560/213841865-caa302f7-66ba-4608-8fad-dc6b40b744e9.png)

**Reply by chadwhitacre** [reply]
> Friendly JSON view

[Browser plugin](https://chrome.google.com/webstore/search/json?_category=extensions)? Solves the problem across the whole World Wide Web. :-)

**Reply by therealarkin** [reply]
Thanks, @mwarkentin! We really appreciate it!

A couple of follow-ups: 
1. What do you hope to gain by zooming in? Let's say you could, how would it help you? And why?
2. JSON - 100%. When do you find yourself needing to look into the JSON? And Why?
3. Event Grouping, thanks! @gaprl - can we add this to our next polish sprint?
4. Tags - we are looking for ways to overhaul the tags cloud. What are the most helpful tags and why?

**Reply by mwarkentin** [reply]
@therealarkin 

1. Zooming in really just lets you pinpoint things more accurately. When I'm at a 24h view and there was a 1h period (or 5m or 1m) where there was a large spike in errors, I don't really have a way to filter down to just that time frame to try to understand if there was an issue with a specific browser type or user or something else.
2. I can't say I've had a use case, but we do offer it up pretty prominently at the top of our issue and it's a bit jarring to end up on a raw, unformatted JSON page
3. Thanks!
4. I don't use tags too heavily currently so don't have too much feedback on the usefulness of them, was more a note about confusion between the tags and context. I also feel like the tag cloud is quite "busy" visually, I wonder if there would be ways to make it easier to scan by aligning things in some way.

**Reply by mwarkentin** [reply]
I just realized that I can zoom in on the issue list page but not on the issue details - this is a good example of where it can show how an issue is reacting to a fix which is hard to see in the issue details page:

![image](https://user-images.githubusercontent.com/67560/220166805-8dab2b56-c4e9-485b-badf-2ca3e93024f5.png)

![image](https://user-images.githubusercontent.com/67560/220167016-8603770c-5784-4af2-9d47-7caf0d0a6448.png)

**Reply by shaedrich** [reply]
Tags like `organisation` and `organisation.slug` could be combined into one until hovered/expanded. Would that help?

**Comment by therealarkin** [comment]
This is one related feedback by @JLuse: https://github.com/getsentry/sentry/issues/44276

**Comment by maxkosty** [comment]
More stuff ≠ more value.

@robinrendle I made an honest effort to remove anything prescriptive unless it was screamingly obvious. Still got left with a lot of those. Please explore the full space of solutions and prove me wrong. 

![issue page feedback](https://user-images.githubusercontent.com/490201/223884376-800c3b39-069c-432e-a44b-0615c18d0cab.png)

**Reply by maxkosty** [reply]
Ben Peven just helped get some data validating **4.d**:

> In the past month only 9-10% of visitors have click either one of the tabs at the top or the tag section above the fold. Comparatively, 26% of visitors clicked into the stack trace

**Reply by maxkosty** [reply]
Another option for simplifying tabs would be to merge User Feedback, Attachments and Replays into a view that has 4 tabs/filters:

```
| All | Replays | User Feedback | Attachments |
```

`All` would be selected by default. Counter badge on the top level count would reflect total count so it would show (11) say if there are 1 Replay, 10 Attachments and 0 User Feedback.

The problem is what to call the top-level tab then. "Attachments" would be a good one but is already taken by attachments 🙂. Artifacts? Add-ons? Extras?

**Reply by chadwhitacre** [reply]
"Additional Context"?

**Reply by shaedrich** [reply]
Some buttons might be combined into a dropdown button like this:

**Issues**
* Merged Issues
* Similar Issues
* Older
* Newer

**Analyze**
* Replay
* Open in Discover

**Comment by therealarkin** [comment]
Hi,

Three updates I'd like to share:
1. We improved SQL readability [read more [here](https://changelog.getsentry.com/announcements/improved-sql-readability)]
2. We improved Usability and Legibility of stack traces [read more [here](https://changelog.getsentry.com/announcements/improved-usability-and-legibility-of-stack-traces)]
3. We made the page load a lot faster when opening it up from the stream

**Comment by rachrwang** [comment]
I'm excited to announce that we're rolling out major updates to early adopters over the next few days, focused on streamlining the debugging workflow and making it easier to access critical information.

![image](https://github.com/user-attachments/assets/36b5b4f3-9f7c-42f8-9643-9016b7312bec)

Learn more here: https://github.com/getsentry/sentry/discussions/80801
